---
title: "Single cell tutorial"

output: 
  github_document:
    toc: true
date: "2024-11-19"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


```{r}
#devtools::install_github("mamarkevi/plotCoverage")
library('plotCoverage')
```

# Download GBM data from 10x
```{bash echo=TRUE, cache=TRUE, results='hide'}
path="https://s3-us-west-2.amazonaws.com/10x.files/samples/cell-vdj/4.0.0/Parent_SC5v1_Human_Glioblastoma/Parent_SC5v1_Human_Glioblastoma"
wget -nv ${path}_possorted_genome_bam.bam -O alignment.bam
wget -nv ${path}_possorted_genome_bam.bam.bai -O alignment.bam.bai
wget -nv ${path}_analysis.tar.gz -O clusters.tar.gz
tar -xzf clusters.tar.gz
```

# Download gene annotation (2020A)
```{bash cache=TRUE, echo=TRUE, include=FALSE}
wget -nv http://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_44/gencode.v44.primary_assembly.annotation.gtf.gz -O genes.gtf.gz
# subset for gene of interest to make things faster
# gene was selected manually
zgrep ENSG00000092841 genes.gtf.gz > gene.gtf
```
# Plot per celltype coverage
## load celltype annotation
```{r}
celltypes = read.csv('analysis/clustering/graphclust/clusters.csv')
```
## loag gene annotation
```{r cache=TRUE}
# it can take a while (5-10min)
gtf = plotCoverage::loadEnsGTF('gene.gtf')
# set colors for exons
# mark exon of interest in red
gtf$exon.col = 'black'
gtf$cds.col = 'black'
f = gtf$start >= 56160500 & gtf$stop <= 56160700
gtf$exon.col[f]=gtf$cds.col[f] = 'red'
```

## load coverage
```{r}
cts = sort(unique(celltypes$Cluster))
cov = lapply(cts,function(cl){
  getReadCoverage('alignment.bam',
                  chr = 'chr12',
                  start = 56160250,
                  end = 56161500,
                  scanBamFlags=list(isNotPassingQualityControls=FALSE,isDuplicate=FALSE,isSupplementaryAlignment=FALSE,isSecondaryAlignment=FALSE),
                  tagFilter = list('CB'=celltypes$Barcode[celltypes$Cluster==cl]) # cellect barcodes for given selltype
                  )
})
names(cov) = cts
```

## plot
```{r fig.height=10, fig.width=4}
# select celltypse to show
# from high to low PSI 
cl_order = as.character(c(6,2,4,1,10,5))
layout(as.matrix(1:(length(cl_order)+1),ncol=1),heights = c(rep(1,length(cl_order)),2))
par(bty='n',tcl=-0.2,mgp=c(1.3,0.3,0),mar=c(0,4,1.5,0),oma=c(0,0,3,1))
for(cl in cl_order)
  plotReadCov(cov[[cl]],main=cl,xaxt='n',ylab='Coverage',min.junc.cov = 2,min.junc.cov.f = 0.02)
plotTranscripts(gtf,new = T,exon.col = NA,cds.col = NA,xlim=c(cov[[1]]$start,cov[[1]]$end))
```
